crypto/stack/stack.c

```diff
@@ -123,6 +123,19 @@ OPENSSL_STACK *OPENSSL_sk_new(OPENSSL_sk_compfunc c)
     return ret;
 }
 
+OPENSSL_STACK *OPENSSL_sk_new_reserve(OPENSSL_sk_compfunc c, int n)
+{
+    OPENSSL_STACK *st;
+
+    if ((st = OPENSSL_sk_new(c)) == NULL
+         || !OPENSSL_sk_reserve(st, n)) {
+        OPENSSL_sk_free(st);
+        return NULL;
+    }
+
+    return st;
+}
```

Comment:
Well, suggestion was rather to arrange it so that OPENSSL_sk_new(c) would call OPENSSL_sk_new_reserve(c,0) [not vice versa], and OPENSSL_sk_new_null would call OPENSSL_sk_new_reserve(NULL,0).

Reply:
I'm with @dot-asm here.